packages/account/src/providers/provider.ts

```diff
@@ -471,16 +471,20 @@ export default class Provider {
    * @hidden
    */
   protected constructor(url: string, options: ProviderOptions = {}) {
-    const { url: rawUrl, urlWithoutAuth, headers } = Provider.extractBasicAuth(url);
+    const { url: rawUrl, urlWithoutAuth, headers: authHeaders } = Provider.extractBasicAuth(url);
 
     this.url = rawUrl;
     this.urlWithoutAuth = urlWithoutAuth;
-    this.options = { ...this.options, ...options };
     this.url = url;
 
-    if (headers) {
-      this.options = { ...this.options, headers: { ...this.options.headers, ...headers } };
-    }
+    const { FUELS } = versions;
+    const headers = { ...authHeaders, ...options.headers, Source: `ts-sdk-${FUELS}` };
```

Comment:
```suggestion
    const headers = { ...authHeaders, Source: `ts-sdk-${FUELS}`, ...options.headers };
```

Maybe allowing for overrides here would be beneficial, then we could also determine if say it's a connector/app?

Reply:
They can always add their own header here, e.g. `source-connector: 'best-connector-ever'`.